src/services/AccessTokenRequestProcessor.ts

```diff
@@ -0,0 +1,127 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { MetricUnits, Metrics } from "@aws-lambda-powertools/metrics";
+import { KmsJwtAdapter } from "../utils/KmsJwtAdapter";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { Response } from "../utils/Response";
+import { ISessionItem } from "../models/ISessionItem";
+import { absoluteTimeNow } from "../utils/DateTimeUtils";
+import { Constants, EnvironmentVariables } from "../utils/Constants";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { AppError } from "../utils/AppError";
+import { BavService } from "./BavService";
+import { checkEnvironmentVariable } from "../utils/EnvironmentVariables";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { isPayloadValid, validateTokenRequestToRecord } from "../utils/AccessTokenRequestValidations";
+
+export class AccessTokenRequestProcessor {
+	private static instance: AccessTokenRequestProcessor;
+
+	private readonly logger: Logger;
+
+	private readonly metrics: Metrics;
+
+	private readonly bavService: BavService;
+
+	private readonly kmsJwtAdapter: KmsJwtAdapter;
+
+	private readonly issuer: string;
+
+	constructor(logger: Logger, metrics: Metrics) {
+		this.logger = logger;
+		this.metrics = metrics;
+  		logger.debug("metrics is  " + JSON.stringify(this.metrics));
+  		this.metrics.addMetric("Called", MetricUnits.Count, 1);
+
+		const sessionTableName: string = checkEnvironmentVariable(EnvironmentVariables.SESSION_TABLE, this.logger);
+  		const signingKeyArn: string = checkEnvironmentVariable(EnvironmentVariables.KMS_KEY_ARN, this.logger);
+		this.issuer = checkEnvironmentVariable(EnvironmentVariables.ISSUER, this.logger);
+		
+		this.kmsJwtAdapter = new KmsJwtAdapter(signingKeyArn);
+		this.bavService = BavService.getInstance(sessionTableName, this.logger, createDynamoDbClient());
+	}
+
+	static getInstance(logger: Logger, metrics: Metrics): AccessTokenRequestProcessor {
+		if (!AccessTokenRequestProcessor.instance) {
+			AccessTokenRequestProcessor.instance = new AccessTokenRequestProcessor(logger, metrics);
+		}
+		return AccessTokenRequestProcessor.instance;
+	}
+
+	async processRequest(event: APIGatewayProxyEvent): Promise<Response> {		
+		try {
+			let requestPayload;
+			try {
+				requestPayload = isPayloadValid(event.body);
+			} catch (error) {
+				this.logger.error("Failed validating the Access token request body.", { messageCode: MessageCodes.FAILED_VALIDATING_ACCESS_TOKEN_REQUEST_BODY });
```

Comment:
Do you think it would be helpful to also log error? We know it won't have an PII and it could help to debug